examples/routing/src/Main.scala

```diff
@@ -0,0 +1,127 @@
+package myorg
+
+import cats.effect.IO
+import org.scalajs.dom
+import tyrian.Html.{param => _, _}
+import tyrian.*
+import urldsl.language.simpleErrorImpl.*
+import urldsl.vocabulary.Param
+import urldsl.vocabulary.Segment
+import urldsl.vocabulary.UrlMatching
+
+import scala.scalajs.js.annotation.JSExportTopLevel
+
+@JSExportTopLevel("TyrianApp")
+object HelloTyrian extends TyrianApp[Msg, Model]:
+
+  val homePath        = root / "home"
+  val counterPath     = root / "counter"
+  val moreComplexPath = root / "id" / segment[Int]
+  val pathWithParam   = (root / "user" / endOfSegments) ? param[Int]("age").?
+
+  private def getPageFromURL(path: String): Page =
+    homePath
+      .matchRawUrl(path)
+      .fold(
+        _ =>
+          counterPath
+            .matchRawUrl(path)
+            .fold(
+              _ =>
+                moreComplexPath
+                  .matchRawUrl(path)
+                  .fold(
+                    _ =>
+                      pathWithParam
+                        .matchRawUrl(path)
+                        .fold(
+                          _ => Page.NotFound,
+                          { case UrlMatching(_, ageOption) =>
+                            Page.UserAgePage(ageOption)
+                          }
+                        ),
+                    userId => Page.UserPage(userId)
+                  ),
+              _ => Page.Counter
+            ),
+        _ => Page.Home
+      )
```

Comment:
I have to say I find this syntax really unfriendly and hard to read - I couldn't recommend it to anyone.

What I would want to see is something like (totally made up):

```scala
Path(path) match
  case root => Page.Home
  case root / "counter" => Page.Counter
  case _ => Page.NotFound
```

I've had a quick look at the library you've pulled in and the closest I can find is this: https://github.com/sherpal/url-dsl/blob/master/url-dsl/src/test/scala/urldsl/examples/RouterUseCaseExample.scala#L71

But here, they're actually making up a mini router of there own in order to get the syntax they want.